### PR TITLE
Refactor/stream store

### PR DESCRIPTION
**Pull request by gsheasby**
Refactor-only.
Covered by tests in StreamTest.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1219)

### CODE REVIEW

File: atlasdb-client/src/main/java/com/palantir/atlasdb/stream/AbstractGenericStreamStore.java [hunk 1]
```diff
@@ -111,61 +91,82 @@ public final InputStream loadStream(Transaction t, final ID id) {
...
         } catch (IOException e) {
             log.error("Could not create temp file for stream id " + id, e);
             throw Throwables.rewrapAndThrowUncheckedException("Could not create file to create stream.", e);
         }
     }
 
-    private StreamMetadata checkStreamStored(ID id, StreamMetadata metadata) {
+    private void checkStreamStored(ID id, StreamMetadata metadata) {
         if (metadata == null) {
             log.error("Error loading stream " + id + " because it was never stored.");
```
Comment (schlosna) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/stream/AbstractGenericStreamStore.java]: These should be slf4j message patterns

File: atlasdb-client/src/main/java/com/palantir/atlasdb/stream/AbstractGenericStreamStore.java [hunk 2]
```diff
@@ -111,61 +91,82 @@ public final InputStream loadStream(Transaction t, final ID id) {
...
             throw new IllegalArgumentException("Unable to load stream " + id + " because it was never stored.");
         } else if (metadata.getStatus() != Status.STORED) {
             log.error("Error loading stream " + id + " because it has status " + metadata.getStatus());
```
Comment (schlosna) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/stream/AbstractGenericStreamStore.java]: Slf4j message patterns

File: atlasdb-client/src/main/java/com/palantir/atlasdb/stream/AbstractGenericStreamStore.java [hunk 3]
```diff
@@ -111,61 +91,82 @@ public final InputStream loadStream(Transaction t, final ID id) {
...
 
-    protected abstract void loadSingleBlockToOutputStream(Transaction t, ID streamId, long blockId, OutputStream os);
+    protected abstract void loadSingleBlockToOutputStream(Transaction tx, ID streamId, long blockId, OutputStream os);
+
+    private StreamMetadata getMetadata(Transaction transaction, ID id) {
+        return Iterables.getOnlyElement(getMetadata(transaction, Sets.newHashSet(id)).entrySet()).getValue();
+    }
 
-    protected abstract Map<ID, StreamMetadata> getMetadata(Transaction t, Set<ID> streamIds);
+    protected abstract Map<ID, StreamMetadata> getMetadata(Transaction tx, Set<ID> streamIds);
```
Comment (SerialVelocity) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/stream/AbstractGenericStreamStore.java]: Can we group the abstract methods?

Comment (gsheasby) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/stream/AbstractGenericStreamStore.java]: Yes - we *also* have to group the overridden methods...

Comment (gsheasby) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/stream/AbstractGenericStreamStore.java]: Done

**Pull request merged**